cou/apps/app.py

```diff
@@ -394,6 +394,18 @@ def post_upgrade_plan(self, target: OpenStackRelease) -> list[UpgradeStep]:
         """
         return [self._get_reached_expected_target_plan(target)]
 
+    def upgrade_description(self, target: OpenStackRelease) -> str:
+        """Top description for upgrading the application.
```

Comment:
Not critical: I think this level of function splitting is overkill.

Reply:
I also agree that this is over-engineering

Reply:
I think it's ok since it can be overwrite during inherited, as it's done in  this [line](https://github.com/canonical/charmed-openstack-upgrader/pull/42/files#diff-a29275aa9b0cacf5c69098e1afcce54ff8c5c0c6cd0480d12a5b483939047b34R40). It's not used only in one place.